Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs

```diff
@@ -2621,13 +2622,21 @@ private async Task<bool> PrepareDirectoryOutputsAsync(HashSet<AbsolutePath> pres
                                         }
                                     });
 
+                                bool allOutputsPrivate = true;
+
                                 foreach (var path in filePaths)
                                 {
                                     if (!await m_makeOutputPrivate(path))
                                     {
-                                        throw new BuildXLException("Failed to create a private, writeable copy of an output file from a previous invocation.");
+                                        allOutputsPrivate = false;
+                                        break;
                                     }
                                 }
+
+                                if (!allOutputsPrivate)
+                                {

```

Comment:
Suggest we log something to indicate preserving outputs failed and we had to fall back on deleting them instead